okapi/services/attrs/attribute-definitions.xml

```diff
@@ -3976,4 +4024,137 @@ for the original implementations.
         </lang>
     </attr>
 
+    <attr acode="A78" categories="de-cache-types">
+        <ocgs id="216" inc="true" />
```

Comment:
So far I have packed OCGS IDs into the numberspace 106-156, and would have used 159+ if none of those are left. (28, 129, 136, 138, 139, 140, 144, 146, 148, 151, 152 still had been free.)

I would prefer to keep the OCGS numberspace as compact as possible, because we are cannibalizing Groundspeak numberspace here. GS some time may need more then 105 different attributes, which would make new GS attrib IDs collide with OCGS IDs. Then we will need to negotiate with GS so that they skip OCGS numberspace. Negotiations may be easier if we did our best to keep OC footprint in their numberspace small.

(Of course, it would have been much better if the inventor of that OCGS number scheme would have started at 1000 rather than 100, avoiding collisions if Groundspeak adds some dozend new attribs. But he did not... and ended up with OCGS IDs in the range 106 - 158. So I stayed with that range for backward compatibility with OC.DE site.

Reply:
On the other hand, we could go directly to 1000 based OC attribs from here on (and also keeping the 9000 convention for negated attributes). This would make it very unlikely to overlap with GS IDs in the future.

Another important concept in bringing order to much existing almost chaos was to unify all [most] numberings. OKAPI has it's own aggregation and is also a GPX generator source (AFAIK). So ACODES are the second numbering. 

My proposal (next stage) was towards OCDE to switch to the same numberings. (AFAIK easier, since most changes happen in database and little in the code)
And to all third party apps to take into account the new and unified numbering scheme, for the first time (AFAIK) well documented. 

I am not sure [at this time] which/how many apps make actual use of OCGS elements in GPX to assess the potential impact.
I do know some apps (c:geo being one of them) make actual use of OKAPI ACODES.

A transitional period would be inevitable, with apps updating to the new numbering, leaving only old, unmaintained apps or outdated installations still using the previous numbering schemes.

Reply:
Regarding potential negotiations: I seriously doubt GS would even take the time to listen...

Reply:
In light of this discussion I am strongly leaning towards making the ID redefinitions 1000 based.